.github/actions/label/build.js

```diff
@@ -30,29 +30,24 @@ const skip = async ({ core, github }) => {
     ref: REF,
   });
 
-  core.info(`check runs: ${check_runs}`);
-
   const runs = linux
     ? check_runs.filter(
       (run) => run.name === "build" || run.name === "build / linux"
     )
     : check_runs.filter((run) => run.name === "build / macos-x86");
 
-  // Skip this action by default.
-  let skipped = false;
+  let skipAction = false;
   for (run of runs) {
-    // Process this action only if the previous build has been skipped.
+    // If there is already a build, skip this action.
     if (
-      (run.name === "build" && run.conclusion === "skipped")
-    )
-      skipped = true;
-
-    // If there is already a build, skip this action without more conditions.
-    if (run.name === "build / linux" || run.name === "build / macos-x86")
+      run.name === "build / linux"
+      || run.name === "build / macos-x86"
+      || (run.name === "build" && run.conclusion !== "skipped")) {
       return true;
+    }
   }
 
-  return !skipped;
```

Comment:
#3474 not triggering build again, could be caused by we skipping this action by default, and only runs it on finding `build (status: skipped)`, which missed a situation that skipping `build` from github CI processes after this label CI.

instead, we skip build only on build has been processed without any other conditions, plus removing the confused variable `skipped`.